app.rb

```diff
@@ -1,8 +1,46 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+require './lib/player'
+require './lib/mechanics'
+
 class RockPaperScissors < Sinatra::Base
+  configure :development do
+      register Sinatra::Reloader
+    end
+
   get '/test' do
     'test page'
   end
 
+  get '/' do
+    erb(:index)
+  end
+
+  post '/name' do
+    $player = Player.new(params[:player_name])
+    redirect "/play"
+  end
+
+  get '/play' do
+    @player = $player
+    @play = $play
+    erb(:play)
+  end
+
+  post '/choice_rock' do
```

Comment:
This code is so clean and easy to read! One thing you could think about is having one post request for 'choice' and passing in the parameter of rock, paper or scissors

Reply:
That could tie in with my problem with the tests, i had it go to seperate pages so that i could test that it went to the correct end point. When i tried having the tests against the random choice i stuggled. Need to read up on that srand as that sounded like a fix and then only one post of choice instead of all three :)